c/misra/src/rules/RULE-7-4/StringLiteralAssignedToNonConstChar.ql

```diff
@@ -0,0 +1,75 @@
+/**
+ * @id c/misra/string-literal-assigned-to-non-const-char
+ * @name RULE-7-4: A string literal shall only be assigned to a pointer to const char
+ * @description Assigning string literal to a variable with type other than a pointer to const char
+ *              and modifying it causes undefined behavior .
+ * @kind problem
+ * @precision very-high
+ * @problem.severity error
+ * @tags external/misra/id/rule-7-4
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+
+class NonConstCharStarType extends Type {
+  NonConstCharStarType() {
+    this instanceof CharPointerType and
```

Comment:
This doesn't exclude completely other types to which a string can be assigned. For example, a short pointer.

This also makes we wonder how we should address wide characters.
For example, `const wchar_t* = L"foo";`

Reply:
👍 Added support for wide strings.